source/common/common/fine_grain_logger.cc

```diff
@@ -64,17 +69,11 @@ bool FineGrainLogContext::setFineGrainLogger(absl::string_view key, level_enum l
 void FineGrainLogContext::setDefaultFineGrainLogLevelFormat(spdlog::level::level_enum level,
                                                             const std::string& format)
     ABSL_LOCKS_EXCLUDED(fine_grain_log_lock_) {
-  if (level == Logger::Context::getFineGrainDefaultLevel() &&
-      format == Logger::Context::getFineGrainLogFormat()) {
-    return;
-  }
-  absl::ReaderMutexLock l(&fine_grain_log_lock_);
-  for (const auto& it : *fine_grain_log_map_) {
-    if (it.second->level() == Logger::Context::getFineGrainDefaultLevel()) {
-      // if logger is default level now
-      it.second->set_level(level);
-    }
-    it.second->set_pattern(format);
+  absl::WriterMutexLock wl(&fine_grain_log_lock_);
```

Comment:
this is only a writer lock because we modify the verbosity_default_level, perhaps we can transition it to a reader lock when doing the loop to avoid blocking logging from occuring during this call

Reply:
Good point! Done.